src/hardware/shared_adc.rs

```diff
@@ -0,0 +1,113 @@
+/// Shared Internal ADC Support
```

Comment:
As a general comment, I think this whole module could be broken out into a separate crate for other users.

Reply:
Like shared-bus? Or a specific shared-adc?

Reply:
I think we could make a `shared-adc` crate, although I don't know if there's a ton of interest or not.

Reply:
Ok.